### ISSUE TITLE
Allow marking units as spoilery

### ISSUE THREAD

**Issue by CelticMinstrel** [wesnoth/wesnoth#2559]
The units database on the Wesnoth site is a nice place to see a list of all units that exist in a given campaign or era. It also offers ways to hide specific units from the list, using `hide_help` or `do_not_list`. As I understand it, this is generally used for units that are only used for internal mechanical purposes, such as the fog clearer.

However, recently it has started to be used to hide spoiler-inducing units from the units database. In my opinion, this starts to defeat the point of the units database, which is to be, essentially, a database of all units. At the same time, I can approve of the goal of wanting to avoid giving away spoilers.

I think a better solution would be to offer _another_ key in `[unit_type]` which is used only by the units database parser. This would mark the unit as a "spoiler" unit, which should by default be hidden in the units database display. The units database website could then add a button or other mechanism to allow you to show spoiler units. I think this would be a good compromise.

In summary, my proposal is:
* Add a key to `[unit_type]` which marks the unit as a spoiler unit
* Add a button or other UI element on the website to show these spoiler units, which are hidden by default.

**Comment by ProditorMagnus** [comment]
This is already possible, #ifdef \_\_WMLUNITS__

**Comment by ProditorMagnus** [comment]
Some units already have different values for database only https://github.com/wesnoth/wesnoth/blob/9bfd432abef371c6c20d526fe05755fe310ba50d/data/core/units/monsters/Fire_Ant.cfg#L13

**Comment by CelticMinstrel** [comment]
The goal is to avoid spoilers while still allowing the unit to be viewed on units.wesnoth.org, but that approach prevents it from ever showing up.

**Comment by cooljeanius** [comment]
Yeah, I'd look to existing implementations of spoiler tags on various websites/apps (e.g. Wesnoth forums, Discord, Mastodon) for inspiration; usually being "spoilery" means that the content is still there, just obfuscated in some way, so that people have to click through to confirm that they actually want to see it

**Comment by Slayer95** [comment]
I am thinking of "flipping" the spoiler unit cards in u.w.o. so that the back of the card shows a customizable message: <i>Spoiler: This unit will be revealed in Chapter 2 of the campaign.</i>, or <i>Spoiler: Travelers have long talked about horrors on the far north.</i>.

But that sort of thing can be useful for the in-game help too, right? At least for me, it has always been a pain point that advancements get hidden every time Wesnoth is updated (or who knows when that happens).

**Comment by Pentarctagon** [comment]
There's an option in the advanced preferences to show hidden advancements. What units you've seen should also be stored in the preferences, which shouldn't get wiped when Wesnoth updates.

**Comment by Slayer95** [comment]
Still, if there were spoiler-hidden Wesnothpedia entries in-game, new players could look at them, and get a sense of their game progression, as well as eagerness to keep playing to discover them.

Note also that if new players are the target, going through advanced preferences would be a no-go.
